opensplice_cmake_module/env_hook/opensplice.sh.in

```diff
@@ -1,16 +1,29 @@
 export OSPL_HOME=@OpenSplice_HOME@
 
+_DEFAULT_OSPL_URI="file://@CMAKE_INSTALL_PREFIX@/share/opensplice_cmake_module/config/ros_ospl.xml"
+
 if [ -f "$OSPL_HOME/release.com" ]; then
+  _ORIGINAL_OSPL_URI="$OSPL_URI"
   . "$OSPL_HOME/release.com" > /dev/null
-else
-  export OSPL_TMPL_PATH=$OSPL_HOME/etc/opensplice/idlpp
-  _DEFAULT_OSPL_URI=file://$OSPL_HOME/etc/opensplice/config/ospl.xml
-  if [ -z "$OSPL_URI" ]; then
-    export OSPL_URI=$_DEFAULT_OSPL_URI
-  elif [ "$OSPL_URI" != "$_DEFAULT_OSPL_URI" ]; then
-    echo "Warning: OSPL_URI was already set to [[$OSPL_URI]]. This will not override it to the default [[$_DEFAULT_OSPL_URI]]. Please make sure this is the config that you want." 1>&2
+  if [ -n "$_ORIGINAL_OSPL_URI" ]; then
+    export OSPL_URI="$_ORIGINAL_OSPL_URI"
+  else
+    export OSPL_URI="$_DEFAULT_OSPL_URI"
   fi
-  unset _DEFAULT_OSPL_URI
+  unset _ORIGINAL_OSPL_URI
+fi
+
+if [ -z "$OSPL_URI" ]; then
+  export OSPL_URI="$_DEFAULT_OSPL_URI"
+elif [ "$OSPL_URI" != "$_DEFAULT_OSPL_URI" ]; then
+  echo "Warning: OSPL_URI was already set to [[$OSPL_URI]]. This will not override it to the default [[$_DEFAULT_OSPL_URI]]. Please make sure this is the config that you want." 1>&2
+fi
+
+unset _DEFAULT_OSPL_URI
+
+# If release.com was not sourced...
+if [ -z "$OSPL_TMPL_PATH" ]; then
+  export OSPL_TMPL_PATH=$OSPL_HOME/etc/opensplice/idlpp
```

Comment:
What about the lines that follow?

Before there was just if and then else. Now there is if then else, then something that has to be done in either case and then the rest (which used to be in the else case). I'll have to see if I can reorganize it to be in the origin else case.